### PR TITLE
fix: Can't exit previewer through Android back button 

### LINKED ISSUE #15884: [BUG]: Can't exit previewer through Android back button
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

- Open the Browser
- Open the menu and select Preview
- Click the Android back button (the one highlighted by yellow in the screenshot)

### Expected behaviour

The previewer is closed.

### Actual behaviour

Nothing happens on clicking the back button.

### Debug info

```text
2.18 alpha 6
```

### (Optional) Anything else you want to share?

The previewer can be closed by clicking the back button at the top left.

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
## Fixes
* Fixes #15884

## Approach
handle the key event

## How Has This Been Tested?
Unit tested

## Learning (optional, can help others)
Our testers are awesome

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by BrayanDSO — APPROVED** [review]
Had noticed the same before going to sleep yesterday. It's very nice to have it already fixed

File: AnkiDroid/src/test/java/com/ichi2/anki/previewer/PreviewerFragmentTest.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            previewerIdsFile = PreviewerIdsFile(createTransientDirectory(), note.cardIds(col)),
+            currentIndex = 0
+        )
+
+        ActivityScenario.launch<PreviewerActivity>(intent).use { scenario ->
+            scenario.moveToState(Lifecycle.State.RESUMED)
+            scenario.onActivity { previewer ->
+                assertThat("Activity is not finishing", !previewer.isFinishing)
+                // this needs to test the keypress, not the dispatcher
+                previewer.pressPhysicalBackButton()
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/previewer/PreviewerFragmentTest.kt:48]: We could use `Espresso.pressBack()` here.

**Review by lukstbit — APPROVED** [review]
LGTM

File: AnkiDroid/src/test/java/com/ichi2/anki/previewer/PreviewerFragmentTest.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/anki/previewer/PreviewerFragmentTest.kt]: Thanks so much!

**Pull request merged**